tests/pybullet_helpers/ikfast/test_load.py

```diff
@@ -0,0 +1,124 @@
+"""Tests for IKFast loading and installation module."""
+import os
+from unittest.mock import call, patch
+
+import pytest
+
+import predicators.pybullet_helpers.ikfast.load
+from predicators.pybullet_helpers.ikfast import IKFastInfo
+from predicators.pybullet_helpers.ikfast.load import \
+    install_ikfast_if_required, install_ikfast_module
+from predicators.utils import get_third_party_path
+
+_MODULE_PATH = predicators.pybullet_helpers.ikfast.load.__name__
+
+
+@pytest.fixture(scope="module", name="ikfast_info")
+def _ikfast_info_fixture() -> IKFastInfo:
+    return IKFastInfo(
+        module_dir="cool_robot",
+        module_name="ikfast_cool_robot",
+        base_link="cool_link0",
+        ee_link="cool_link7",
+        free_joints=["cool_link6"],
+    )
+
+
+def test_install_ikfast_module():
+    """Test for install_ikfast_module."""
+    with patch(f"{_MODULE_PATH}.os") as mock_os:
+        mock_os.system.return_value = 0
+        install_ikfast_module("/path/to/ikfast")
+        mock_os.system.assert_called_once_with(
+            "cd /path/to/ikfast; python setup.py")
+
+
+@pytest.mark.parametrize("exit_code", list(range(1, 5)))
+def test_install_ikfast_module_raises_error(exit_code):
+    """Test install_ikfast_module raises error if os.system returns non-
+    zero."""
+    with patch(f"{_MODULE_PATH}.os") as mock_os:
+        mock_os.system.return_value = exit_code
+
+        with pytest.raises(RuntimeError):
+            install_ikfast_module("/path/to/ikfast")
+
+
+def test_install_ikfast_if_required_installs_ikfast_module(ikfast_info):
+    """Test install_ikfast_if_required installs IKFast if there are no existing
+    module files."""
+    expected_ikfast_dir = os.path.join(get_third_party_path(), "ikfast",
+                                       ikfast_info.module_dir)
+    expected_module_path = os.path.join(
```

Comment:
this assumes linux right? that's okay, just checking

Reply:
Yes, but only for the purposes of this test. The file doesn't exist as we're just mocking things so it doesn't really matter